calyx-py/test/red_tree.py

```diff
@@ -0,0 +1,166 @@
+# pylint: disable=import-error
+import calyx.builder as cb
+
+
+def add_adder(
+    comp: cb.ComponentBuilder,
+    adder: cb.CellBuilder,
+    group,
+    port_l,
+    port_r,
+    ans,
+):
+    """To component {comp}, adds wiring for an adder-group called {group}.
+    Assumes the adder cell {adder} is in the component, and puts {port_l} and {port_r}
+    into the adder.
+    Then puts the output of {adder} into the memory register {ans}.
+    Returns the group.
+    """
+    with comp.group(group) as adder_group:
+        adder.left = port_l
+        adder.right = port_r
+        ans.write_en = 1
+        ans.in_ = adder.out
+        adder_group.done = ans.done
+    return adder_group
+
+
+def add_tree(prog):
+    """Inserts the component `tree` into the program.
+    It has:
+    - four inputs, `leaf1`, `leaf2`, `leaf3`, and `leaf4`
+    - one output, `sum`
+
+    When done, it puts the sum of the four leaves into `sum`.
+    Returns the component builder for tree.
+    """
+
+    tree: cb.ComponentBuilder = prog.component("tree")
+    for i in range(1, 5):
+        tree.input(f"leaf{i}", 32)
+    # AM, quality of life:
+    # `input` has no return value, so I'm forced to immediately call `this()`
+    # to get handles to the ports I just created.
+    [leaf1, leaf2, leaf3, leaf4] = [tree.this()[f"leaf{i}"] for i in range(1, 5)]
```

Comment:
@EclecticGriffin @sampsyo 
Just flagging a small quality-of-life thing in the builder library! Please let me know if I'm missing something, and if not, if you'd be interested in making a change!